pkg/apiserver/storage/ram/store_test.go

```diff
@@ -453,3 +455,64 @@ func TestRamStoreWatchWithSelector(t *testing.T) {
 		}
 	}
 }
+
+func TestRamStoreWatchTimeout(t *testing.T) {
+	store := NewStore(cache.MetaNamespaceKeyFunc, cache.Indexers{}, testGenEvent)
+	// watcherChanSize*2+1 events can fill a watcher's buffer: input channel buffer + result channel buffer + 1 in-flight.
+	maxBuffered := watcherChanSize*2 + 1
+
+	// w1 has consumer for its result chan.
+	w1, err := store.Watch(context.Background(), "", labels.SelectorFromSet(labels.Set{"app": "nginx"}), fields.Everything())
+	if err != nil {
+		t.Errorf("Failed to watch object: %v", err)
+	}
+	go func() {
+		ch := w1.ResultChan()
+		for i := 0; i < maxBuffered+1; i++ {
+			actualEvent := <-ch
+			expectedEvent := watch.Event{watch.Added, &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("pod%d", i), Labels: map[string]string{"app": "nginx"}}}}
+			if !reflect.DeepEqual(actualEvent, expectedEvent) {
+				t.Errorf("Unexpected event %d, got %#v, expected %#v ", i, actualEvent, expectedEvent)
+			}
+		}
+		select {
+		case obj, ok := <-ch:
+			t.Errorf("Unexpected excess event: %#v %t", obj, ok)
+		default:
+		}
+	}()
+
+	// w2 has no consumer for its result chan.
+	w2, err := store.Watch(context.Background(), "", labels.SelectorFromSet(labels.Set{"app": "nginx"}), fields.Everything())
+	if err != nil {
+		t.Errorf("Failed to watch object: %v", err)
+	}
+
+	watchersNum := store.GetWatchersNum()
+	if !assert.Equal(t, 2, watchersNum) {
```

Comment:
`assert.Equal(t, 2, watchersNum)` is enough IMO as it will print the statement + the expected value + the actual value.

Reply:
thanks for reminding, will remove the repeated message